grpc.go

```diff
@@ -780,6 +788,19 @@ func (a *agentGRPC) WaitProcess(ctx context.Context, req *pb.WaitProcessRequest)
 		return &pb.WaitProcessResponse{}, err
 	}
 
+	select {
+	//the exitCodeCh is closed, do nothing
+	case <-proc.exitCodeCh.exitCodeCh:
```

Comment:
One `WaitProcess` goroutine can pick up the `exitCode` filled by another `WaitProcess` goroutine and then the exit code is lost forever. All remaining waiting goroutines in `reaper.wait()` will be blocked forever.

It is a hard to test scenario because the check here happens quickly, but the race window is there.

Reply:
No, the other WaitProcess will be blocked on the line of:
exitCode, err := a.sandbox.subreaper.wait(proc.exitCodeCh, libContProcess)
 and the exitCode only will be picked up by it and once they come to the "select" block, 
the exitCode has been picked up there.

Reply:
Good point! Thanks for the explanation!